auto_submit/lib/service/validation_service.dart

```diff
@@ -154,4 +175,46 @@ class ValidationService {
     }
     return true;
   }
+
+  /// The logic for processing a revert request. If the revert request is
+  /// processed successfully a new tracking review issue is opened as a result
+  /// since we want to be able to follow up with a formal review.
+  ///
+  /// TODO (ricardoamador) move this code to its own service.
+  Future<void> processRevertRequest(
+      Config config, QueryResult result, github.PullRequest messagePullRequest, String ackId, PubSub pubsub) async {
+    ValidationResult revertValidationResult = await revertValidation!.validate(result, messagePullRequest);
+    if (revertValidationResult.result) {
+      bool processed = await processMerge(config, result, messagePullRequest);
+      if (processed) {
+        await pubsub.acknowledge('auto-submit-queue-sub', ackId);
+      }
+      log.info('Ack the processed message : $ackId.');
+      github.RepositorySlug slug = messagePullRequest.base!.repo!.slug();
+      final int prNumber = messagePullRequest.number!;
+      final GithubService githubService = await config.createGithubService(slug);
+
+      github.Issue issue = await githubService.createIssue(
+        slug,
+        'Follow up review for revert pull request $prNumber',
+        'Revert request by author ${result.repository!.pullRequest!.author}',
+      );
+
+      log.info('$issue was created to track the review for $prNumber');
+    } else {
+      // since we do not temporarily ignore anything with a revert request we
+      // know we will report the error and remove the label.
+      final int prNumber = messagePullRequest.number!;
+      github.RepositorySlug slug = messagePullRequest.base!.repo!.slug();
+      final GithubService gitHubService = await config.createGithubService(slug);
```

Comment:
Dup logics that can be moved out of the if else.

Reply:
Fixed.